### PR TITLE
adding a method for obtaining the agenda uri

### LINKED ISSUE #43: Portland, OR scraper
### Feature Description

_A clear and concise description of the feature you're requesting._

Create `portland.py` or a similarly named legislation events scraper for Portland, OR.

### Use Case

_Please provide a use case to help us understand your request in context._

The scraper will provide `ingestion_models` for `cdp-backend` when we deploy an instance for the city.

### Solution

_Please describe your ideal solution._

`PortlandScraper.get_events()` return `List[EventIngestionModel]` filled with more-than-minimal information, such as `EventMinutesItem.votes`. Because Portland, OR does not use Legistar, the scraper must obtain all the information by parsing the relevant web page(s).

### Alternatives

_Please describe any alternatives you've considered, even if you've dismissed them._

The `cdp-backend` pipeline will not work without an events scraper. There is no alternative solution.

### PR DESCRIPTION
**Pull request by Shak2000**
### Link to Relevant Issue

This pull request is part of issue #43 

### Description of Changes

This pull request adds a method for obtaining the `agenda_uri` to the Portland scraper as well as a unit test for that.

### CODE REVIEW

**Comment by dphoria** [comment]
Shak while verifying this PR I noticed I had left a bug (as opposed to missing feature) in `portland.py`. I fixed and have merged the bugfix into the repo. Can you please pull `upstream/main` into your fork so that it gets #63 ? Thank you.

File: cdp_scrapers/instances/portland.py [hunk 1]
```diff
@@ -281,3 +281,21 @@ def get_events(
             # for easier iterate there
             collapse=False,
         )
+
+    def get_agenda_uri(self, event_page: BeautifulSoup) -> str:
```
Comment (dphoria) [inline cdp_scrapers/instances/portland.py:285]: Please move this to above the caller, i.e. `get_event()`.

**Comment by Shak2000** [comment]
Done!

**Comment by dphoria** [comment]
This works for your test case of 2021/12/22, but not for others. For example, for 2022/1/12, I am getting `agenda_uri='https://portlandoregon.us16.list-manage.com/subscribe?u=fb4436284b536324673a1c80c&id=1d80c3d2cc/File/Document'`, which results in HTTP 404 error.

If one rule doesn't work, we may need to do something like "try this, or that". Ideally, we can use 1 rule every time, for sure. But at the moment, I'm not confident that is the case, unfortunately.

**Comment by dphoria** [comment]
> ### Link to Relevant Issue
> 
> This pull request resolves #43

By the way Shak, could you change this text to something else? Unlike other issues and PRs, we don't want this PR to close that issue. The scraper will not be finished by this PR alone. e.g. We can use something like

> This pull request is part of #43

or similar. I have seen when I do that, GitHub doesn't automatically link the issue, or close it when this PR is merged. This is a minor point, but does save us the few clicks of reopening the issue after merging this PR.

**Comment by dphoria** [comment]
Looks like it's working perfectly. Thank you Shak! The only remaining nitpick I have is the comment I made previously, about putting `get_agenda_uri()` above the method that calls it, i.e. `get_event()`

https://github.com/CouncilDataProject/cdp-scrapers/pull/62#discussion_r783531758

**Pull request merged**